drivers/SmartThings/matter-thermostat/src/init.lua

```diff
@@ -358,7 +369,19 @@ local function do_configure(driver, device)
     device.log.info_with({hub_logs=true}, string.format("Updating device profile to %s.", profile_name))
     device:try_update_metadata({profile = profile_name})
   elseif device_type == FAN_DEVICE_TYPE_ID then
-    device.log.warn_with({hub_logs=true}, "Fan supports only one profile")
+    profile_name = "fan"
+    if #wind_eps > 0 then
+      profile_name = profile_name .. "-wind"
+    end
+    if #rock_eps > 0 then
+      profile_name = profile_name .. "-rock"
+    end
+    if profile_name == "fan" then
```

Comment:
Why not default `profile_name` to "fan-generic" up above and then you can remove this `if`

Reply:
If we did this, we'd be creating profiles like "fan-generic-wind", which I guess isn't bad sounding now that I think about it

Reply:
🤦 didn't think of that. No, it's better as is